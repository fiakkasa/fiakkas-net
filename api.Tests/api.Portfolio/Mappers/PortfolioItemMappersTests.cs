using api.Portfolio.Interfaces;
using api.Portfolio.Models;

namespace api.Portfolio.Mappers;

public class PortfolioItemMappersTests
{
    public record PortfolioItemMockEntity : IPortfolioItem
    {
        public Guid Id { get; init; }
        public DateTimeOffset CreatedAt { get; init; }
        public DateTimeOffset? UpdatedAt { get; init; }
        public long Version { get; init; }
        public long Year { get; init; }
        public Guid CategoryId { get; init; }
        public long Ordinal { get; init; }
        public string Title { get; init; } = string.Empty;
        public Uri? Href { get; init; }
        public Guid[] TechnologyIds { get; init; } = [];
        public Guid CustomerId { get; init; }
    }

    [Fact]
    public void Map_Should_Return_Data()
    {
        var item = new PortfolioItemMockEntity
        {
            Id = new Guid("28e483e4-6961-4b25-88a9-d1d0a5161109"),
            CreatedAt = new(2024, 1, 1, 0, 0, 0, TimeSpan.Zero),
            UpdatedAt = null,
            Version = 1,
            Year = 2024,
            CategoryId = new Guid("38e483e4-6961-4b25-88a9-d1d0a5161109"),
            Ordinal = 1,
            Title = "Title",
            Href = new Uri("/test", UriKind.Relative),
            TechnologyIds = [new Guid("48e483e4-6961-4b25-88a9-d1d0a5161109")],
            CustomerId = new Guid("18e483e4-6961-4b25-88a9-d1d0a5161109")
        };

        var result = item.Map();

        result.Should().BeOfType<PortfolioItem>();
        result.Should().BeEquivalentTo(item);
        result.MatchSnapshot();
    }
}
