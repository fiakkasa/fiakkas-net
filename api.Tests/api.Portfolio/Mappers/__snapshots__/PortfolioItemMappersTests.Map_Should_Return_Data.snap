{
  "Year": 2024,
  "CategoryId": "38e483e4-6961-4b25-88a9-d1d0a5161109",
  "Title": "Title",
  "Href": "/test",
  "TechnologyIds": [
    "48e483e4-6961-4b25-88a9-d1d0a5161109"
  ],
  "CustomerId": "18e483e4-6961-4b25-88a9-d1d0a5161109",
  "Id": "28e483e4-6961-4b25-88a9-d1d0a5161109",
  "CreatedAt": "2024-01-01T00:00:00+00:00",
  "UpdatedAt": null,
  "Version": 1
}
